### ISSUE TITLE
Cleanup handling of SIMD instructions in the emitter

### ISSUE THREAD

**Issue by tannergooding** [dotnet/coreclr#16144]
Today, the emitter special cases the handling of many SIMD instructions or their special encoding requirements (such as the VEX encoding, which takes 3 operands, or the 4-byte AVX/SSE4.1 encodings).

The hardware intrinsics feature has added direct support for handling many of these same scenarios with dedicated code paths.

It may be beneficial to do a cleanup of this code, in general, and remove the special handling in favor of calling the explicit code paths.

This would likely result in fewer unnecessary checks/handling and provide more maintainable/understandable code.

**Comment by tannergooding** [comment]
FYI. @fiigii, @CarolEidt

**Comment by fiigii** [comment]
related to https://github.com/dotnet/coreclr/issues/14523

**Comment by CarolEidt** [comment]
> related to #14523

Yes - it would be great to see this, as well as the other somewhat convoluted paths in the emitter, handled in a more data-driven fashion.
